File: python-worker/src/al_iyaal_worker/timecode.py
import math


def parse_time_to_seconds(value: str) -> float:
    parts = value.strip().replace(",", ".").split(":")
    if not 1 <= len(parts) <= 3 or any(not part for part in parts):
        raise ValueError("time value is empty")

    values = [float(part) for part in parts]
    if any(not math.isfinite(part) for part in values):
        raise ValueError("time value must be finite")
    if values[0] < 0 or any(part < 0 or part >= 60 for part in values[1:]):
        raise ValueError("time value contains an invalid component")

    total = 0.0
    multiplier = 1.0
    for part in reversed(values):
        total += part * multiplier
        multiplier *= 60.0
    return total

File: src/features/shared/timecode.ts
const parseFlexibleTimeToSeconds = (value: unknown): number | null => {
  if (typeof value === "number") {
    return Number.isFinite(value) && value >= 0 ? value : null;
  }
  if (typeof value !== "string" || !value.trim()) {
    return null;
  }
  const parts = value.trim().replace(",", ".").split(":");
  if (parts.length < 1 || parts.length > 3 || parts.some((part) => !part)) {
    return null;
  }
  const values = parts.map(Number);
  if (values.some((part) => !Number.isFinite(part))) {
    return null;
  }
  if ((values[0] ?? -1) < 0 || values.slice(1).some((part) => part < 0 || part >= 60)) {
    return null;
  }
  return values.reduce((total, part) => total * 60 + part, 0);
};

export { parseFlexibleTimeToSeconds };
